### ISSUE TITLE
Websocket connection failed

### ISSUE THREAD

**Issue by AdriVanHoudt** [adobe-research/theseus#75]
While doing nothing Theseus keeps throwing errors in console
![] (http://puu.sh/eUiTd/553fd525b0.png)
And in text. (1076 is the counter btw)
```javascript
1076 /C:/Users/adri/AppData/Roaming/Brackets/extensions/user/theseus/src/Agent-node.js:61 WebSocket connection to 'ws://localhost:8888/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
```
.

**Comment by Robbson** [comment]
It's the same here. I tested it on two different Windows 7 64 machines and on OSX. It makes debugging using the javascript console useless because it gets floatet with those error messages. Also it seems to slow down Theseus.

**Comment by AdriVanHoudt** [comment]
Yeah apart from being annoying it throwing errors all the time can't be good for performance. If this is intended behaviour it should silently fail

**Comment by alltom** [comment]
Patches welcome! I don't know how to fix this easily.

The smallest-change fix (which I can't figure out) would be to squelch the error somehow, because it's completely benign. However, it happens asynchronously, so there's nothing to try-catch. I search the web every so often for a way and haven't found a fix.

A patch-it-over but still small fix would be to not try connecting when Theseus is disabled. At least then you could disable Theseus to mute these errors. This would be less surprising behavior for the automatic retries anyway.

Eventually I hope to flip the connection so that the debugging server is run in Brackets and the debugged process repeatedly tries to connect to Brackets, instead of the other way around, which is how it works now. Then there'd be no connection failed errors in Brackets ever. :)

**Comment by AdriVanHoudt** [comment]
Apperently those errors are by design https://groups.google.com/a/chromium.org/d/msg/chromium-html5/4G_XqPQQFX8/Yt4YN5DiQ0EJ post is from 2011 though.

I agree that turning it around would be better, any ETA on that? :stuck_out_tongue_winking_eye:

**Comment by vincentf305** [comment]
ETA? I am still seeing this issue

**Comment by alltom** [comment]
AdriVanHoudt + vincentfaller: If you could stop the warnings by temporarily disabling Theseus from the File menu, would that be enough to fix your workflow, or do you need the bigger fix?

The small fix is to just check `main.isEnabled()` [here](https://github.com/adobe-research/theseus/blob/master/src/Agent-node.js#L177), so its ETA would be pretty short, but the larger fix is kinda huge.

(Sorry for the delay. I was away for the summer.)

**Comment by AdriVanHoudt** [comment]
@alltom tbh I use node-inspector now but temp disabling it when you turn of theseus in the menu would indeed be a fine solution for me

**Comment by Eagleheart** [comment]
I am experiencing this same issue. Agent-node.js:61 WebSocket connection to 'ws://localhost:8888/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED repeatedly shows up in my console when theseus is installed, even when no program is running or when theseus is set to off. Is this going to be fixed soon?

**Comment by Michaelvdlnz** [comment]
Fixed for me after reboot.
